### ISSUE TITLE
Grafana Unified Alerts do not work with pre-2.10 Prometheus

### ISSUE THREAD

**Issue by hai-ld** [grafana/grafana#40866]
**What happened**: After adding Prometheus  as a data source, accessing `<grafana_base_url>/alerting/list` displays this error
```
Failed to load rules state from Prometheus: json: cannot unmarshal number into Go struct field Alert.data.groups.rules.alerts.value of type string
```

**What you expected to happen**: Prometheus alerts are displayed

**How to reproduce it (as minimally and precisely as possible)**: using pre-2.10 Prometheus will cause the error, due to this breaking change in Prometheus API https://github.com/prometheus/prometheus/pull/5582

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.2.2 (6232fe07c0)
- Data source type & version: Prometheus v2.5.0
- OS Grafana is installed on: Alpine (using official Docker image)
- User OS & Browser: Ubuntu 20.04, Firefox 93
- Grafana plugins: N/A
- Others: N/A

**Comment by ifrost** [comment]
Looping in @pkolyvas @davkal

**Comment by ifrost** [comment]
@hai-ld are you able to upgrade Prometheus to the latest version? That should fix the problem.

**Comment by ivanahuckova** [comment]
Looping in also @grafana/alerting-squad as the error happens when user uses alert api. Not sure how many user use `pre-2.10` version of Prometheus, but this issue might be more frequent when more people start to use unified alerting.

**Comment by hai-ld** [comment]
@ifrost @ivanahuckova I upgraded to 2.10 which solved this particular error, but there's another error related to alerts categorization (e.g. firing, pending) which is solved when upgrading to 2.20. I don't know which particular version starts introducing the feature, but it looks like [this image](https://user-images.githubusercontent.com/1579471/140458462-6403f5d3-53fe-4349-a93a-0867c6b9348c.png). I'm not opposed to upgrading, and Prometheus seems to have great stability on its backend so that shouldn't be a problem. However, Grafana Unified Alerts has implicit dependencies on specific Prometheus features, and I think it's best if the project could make it explicit, at least in the feature's documentation.

**Comment by armandgrillet** [comment]
Thanks for creating this issue, the Alerting squad will triage it next week. Additional context: https://github.com/prometheus/prometheus/releases/tag/v2.10.0 is from May 2019 and https://github.com/prometheus/prometheus/releases/tag/v2.20.0 is from July 2020.
